### DISCUSSION TITLE
basic docker configuration

### DISCUSSION THREAD

**Discussion by mieliespoor** [renovatebot/renovate#17576] [Request Help] [answered]
Quickly help me with this as I'm somewhat confused with the config.

We need to configure renovate so that it will track the major releases for ubuntu, but also do digest pinning. I have read through https://docs.renovatebot.com/docker/, but I'm not sure I follow exactly what I should do to make all this work. 

At the moment, I see two PR's in one repo:

1. ubuntu:20.04 -> ubuntu:20.10
2. ubuntu:20.04 -> ubuntu:22.10

Since we are currently on ubuntu:20.04, I would prefer to then go to ubuntu:22.04 (LTS), so neither of what renovate offers. Then I would also like Renovate to pin the digest and this is not happening in either of the two PR's. 

Regarding the digest pinning, I've just added `docker:pinDigests` so will wait for Renovate to run in the next hour and see what happens.

This is my current renovate config, so not overly complex:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    "docker:pinDigests"
  ],
  "packageRules": [
    {
      "matchUpdateTypes": [
        "minor",
        "patch",
        "pin",
        "digest"
      ],
      "automerge": true
    }
  ]
}
```

**Comment by mieliespoor** [comment]
Great, so adding `docker:pinDigests` does the digest pinning for me. How do I tell it to only look at stable (LTS) versions?

**Comment by PhilipAbed** [comment, marked as answer]
Usually [ignoreUnstable](https://docs.renovatebot.com/configuration-options/#ignoreunstable) is set to true by default,
and version `ubuntu:22.10` is considered stable.

So what you are looking for is [allowedVersions](https://docs.renovatebot.com/configuration-options/#allowedversions)

**Reply by PhilipAbed** [reply]
look at this example might help you understand what i mean: 
https://github.com/renovatebot/renovate/discussions/17560

**Reply by viceice** [reply]
checkout Ubuntu versioning

**Reply by mieliespoor** [reply]
this link? https://docs.renovatebot.com/modules/versioning/#ubuntu-versioning

**Reply by viceice** [reply]
yes

**Reply by mieliespoor** [reply]
OK, I added this for ubuntu now. I will wait and see what this does...

```
{
      "matchDatasources": ["docker"],
      "matchPackageNames": ["ubuntu"],
      "versioning": "ubuntu"
 }
```

**Reply by mieliespoor** [reply]
If this work, I will follow a similar process for our dotnet alpine images as we will need some rules there to stick to the lts versions microsoft provide.

**Reply by mieliespoor** [reply]
I have the alpine images for dotnet sorted out, with the config below, but still being offered non-lts versions for ubuntu (22.10 instead of 22.04). Not sure what to change.

```json
    {
      "matchDatasources": ["docker"],
      "matchPackageNames": ["ubuntu", "library/ubuntu", "buildimages/library/ubuntu"],
      "versioning": "ubuntu",
      "allowedVersions": "/[0-9]{2}\\.04/"
    },
    {
      "matchDatasources": ["docker"],
      "matchPackagePatterns": "dotnet\/(core\/)?(sdk|aspnet)",
      "allowedVersions": "/(3\\.1|6\\.0)-alpine(-arm64(v8)?)?/"
    }
```

**Reply by rarkins** [reply]
Can you create a minimal reproduction on github.com?

**Reply by mieliespoor** [reply]
That will be a bit more difficult, but I got this from the log:

```
 DEBUG: packageFiles with updates (repository=cicddoc/gitlab.runner.worker.base)
        "config": {
          "dockerfile": [
            {
              "packageFile": "Dockerfile",
              "deps": [
                {
                  "depName": "host.dns/buildimages/library/ubuntu",
                  "currentValue": "20.04",
                  "currentDigest": "sha256:af5efa9c28de78b754777af9b4d850112cad01899a5d37d2617bb94dc63a49aa",
                  "replaceString": "host.dns/buildimages/library/ubuntu:20.04@sha256:af5efa9c28de78b754777af9b4d850112cad01899a5d37d2617bb94dc63a49aa",
                  "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                  "datasource": "docker",
                  "depType": "final",
                  "depIndex": 0,
                  "updates": [
                    {
                      "bucket": "non-major",
                      "newVersion": "20.10",
                      "newValue": "20.10",
                      "newMajor": 20,
                      "newMinor": 10,
                      "updateType": "minor",
                      "newDigest": "sha256:a7b08558af07bcccca994b01e1c84f1d14a2156e0099fcf7fcf73f52d082791e",
                      "branchName": "dependabot/host.dns-buildimages-library-ubuntu-20.x"
                    },
                    {
                      "bucket": "major",
                      "newVersion": "22.10",
                      "newValue": "22.10",
                      "newMajor": 22,
                      "newMinor": 10,
                      "updateType": "major",
                      "newDigest": "sha256:3cc2e19d3538681fb614683187329b47520424b80d7b31c97cdeb2403d82cae9",
                      "branchName": "dependabot/host.dns-buildimages-library-ubuntu-22.x"
                    },
                    {
                      "updateType": "digest",
                      "newValue": "20.04",
                      "newDigest": "sha256:35ab2bf57814e9ff49e365efd5a5935b6915eede5c7f8581e9e1b85e0eecbe16",
                      "branchName": "dependabot/host.dns-buildimages-library-ubuntu-digest"
                    }
                  ],
                  "warnings": [],
                  "versioning": "docker",
                  "currentVersion": "20.04",
                  "isSingleVersion": true,
                  "fixedVersion": "20.04"
                }
              ]
            }
          ]
        }
```

**Reply by viceice** [reply]
you need to match `depName` from this log in your hostrules -> `host.dns/buildimages/library/ubuntu`

**Reply by mieliespoor** [reply]
Ok that help, thanks. I was under the false impression that the dns part of the image name would be ignored.
